pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -118,6 +119,7 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 			FSID:          string(secrets.Data[fsidSecretNameKey]),
 			MonitorSecret: string(secrets.Data[monSecretNameKey]),
 			Context:       context,
+			Spec:          ClusterSpec,
 		}
 		if cephUsername, ok := secrets.Data[cephUsernameKey]; ok {
```

Comment:
> @travisn do we want to update secret `rook-ceph-mon` according to the change in annotation if cephCluster is already created. The reason I'm asking is that we don't update this secret ``secret `rook-ceph-mon` frequently apart from one instance.

Let's go ahead and update the annotations with each reconcile. I believe we can do this simply by adding the annotations in this section after the secret is retrieved. We can add a new block like this down near line 139:

- If there are annotations
   - Apply the annotations to the secret
   - Call `Update()` on the secret

If there are no annotations, no need to update the secret